src/js/form.js

```diff
@@ -95,7 +96,21 @@ Form.prototype = {
             this.required.update.bind(this.required),
             this.readonly.update.bind(this.readonly),
             this.validationUpdate,
-        ].concat(this.evaluationCascadeAdditions);
+        ];
+
+        let evaluationCascade;
+
+        if (config.experimentalOptimizations.computeAsync) {
+            evaluationCascade = baseEvaluationCascade.map(
+                (fn) =>
+                    (...args) =>
+                        callOnIdle(() => fn(...args))
+            );
+        } else {
+            evaluationCascade = baseEvaluationCascade;
+        }
+
+        return evaluationCascade.concat(this.evaluationCascadeAdditions);
```

Comment:
would the (async) `evaluationCascadeAdditions` now run before the (sync) `baseEvalationCascade`? Should this not be concatenated before the if statement?

Reply:
Good catch! Yes I think they should, though probably wrapped together as one function in case there are cases which shouldn't individually be async.